src/AppRun.c

```diff
@@ -25,182 +25,298 @@ THE SOFTWARE.
 
-    // change directory
+    // A trailing null argument is required by `execvp()`.
+    outargptrs[outargindex] = '\0';
+
+    // Change directory to the application directory we found earlier.
+
+    // Calculate the length of the application directory path string.
     size_t appdir_s = strlen(appdir);
+    // Define the path to the ./usr subfolder in the application directory.
     char *usr_in_appdir = malloc(appdir_s + 5);
     snprintf(usr_in_appdir, appdir_s + 5, "%s/usr", appdir);
+    // Attempt to change to the `<appdir>/usr` directory.
     ret = chdir(usr_in_appdir);
-    if (ret != 0)
+    // If we encounter a problem changing to that directory, throw a fatal error.
+    if (ret != 0) {
         die("Could not cd into %s\n", usr_in_appdir);
+    }
 
-    // set environment variables
+    // Append to the environment variables, so the AppImage can run correctly.
     char *old_env;
     size_t length;
     const char *format;
 
-    /* https://docs.python.org/2/using/cmdline.html#envvar-PYTHONHOME */
+    /* Append the Python home path.
+     * https://docs.python.org/2/using/cmdline.html#envvar-PYTHONHOME
+     */
     SET_NEW_ENV(new_pythonhome, appdir_s, "PYTHONHOME=%s/usr/", appdir);
 
+    // Append the system path.
     old_env = getenv("PATH") ?: "";
     SET_NEW_ENV(new_path, appdir_s*5 + strlen(old_env), "PATH=%s/usr/bin/:%s/usr/sbin/:%s/usr/games/:%s/bin/:%s/sbin/:%s", appdir, appdir, appdir, appdir, appdir, old_env);
 
+    // Append the system library path.
     old_env = getenv("LD_LIBRARY_PATH") ?: "";
     SET_NEW_ENV(new_ld_library_path, appdir_s*10 + strlen(old_env), "LD_LIBRARY_PATH=%s/usr/lib/:%s/usr/lib/i386-linux-gnu/:%s/usr/lib/x86_64-linux-gnu/:%s/usr/lib32/:%s/usr/lib64/:%s/lib/:%s/lib/i386-linux-gnu/:%s/lib/x86_64-linux-gnu/:%s/lib32/:%s/lib64/:%s", appdir, appdir, appdir, appdir, appdir, appdir, appdir, appdir, appdir, appdir, old_env);
 
+    // Append the Python path.
     old_env = getenv("PYTHONPATH") ?: "";
     SET_NEW_ENV(new_pythonpath, appdir_s + strlen(old_env), "PYTHONPATH=%s/usr/share/pyshared/:%s", appdir, old_env);
 
+    // Append the XDG path
     old_env = getenv("XDG_DATA_DIRS") ?: "/usr/local/share/:/usr/share/";
     SET_NEW_ENV(new_xdg_data_dirs, appdir_s + strlen(old_env), "XDG_DATA_DIRS=%s/usr/share/:%s", appdir, old_env);
 
+    // Append the Perl path
     old_env = getenv("PERLLIB") ?: "";
     SET_NEW_ENV(new_perllib, appdir_s*2 + strlen(old_env), "PERLLIB=%s/usr/share/perl5/:%s/usr/lib/perl5/:%s", appdir, appdir, old_env);
 
+    // Append the GSettings Schema path
     /* http://askubuntu.com/questions/251712/how-can-i-install-a-gsettings-schema-without-root-privileges */
     old_env = getenv("GSETTINGS_SCHEMA_DIR") ?: "";
     SET_NEW_ENV(new_gsettings_schema_dir, appdir_s + strlen(old_env), "GSETTINGS_SCHEMA_DIR=%s/usr/share/glib-2.0/schemas/:%s", appdir, old_env);
 
+    //Append the Qt Path
```

Comment:
More like the "Qt plugin path". There's other Qt paths, and we specifically augment to the `QT_PLUGIN_PATH` here.